src/dmd/utf.d

```diff
@@ -406,15 +406,15 @@ void utf_encode(int sz, void* s, dchar c)
  * Returns:
  *      null on success, otherwise error message string
  */
-immutable(char*) utf_decodeChar(const(char)* s, size_t len, ref size_t ridx, out dchar rresult)
+string utf_decodeChar(const(char)[] s, ref size_t ridx, out dchar rresult)
```

Comment:
Passing in a slice is fine, but I think the return type could stay a plain pointer: it is a preallocated error message anyway that is simpler printed as a zero-terminated string. It's also a bit less code to check a pointer as a boolean (both pointer and length of a slice are tested).

Reply:
It's a persuasive idea, but I'd really prefer to just get away from 0 terminated strings. The ugly annoyance with `printf` remains, but we can live with that.

Reply:
> The ugly annoyance with `printf` remains, but we can live with that.

I think it's the right time to remind you that [there's a simple solution to that](https://github.com/dlang/dmd/pull/8687).

Reply:
> It's also a bit less code to check a pointer as a boolean (both pointer and length of a slice are tested).

Off-topic, but that's not true, only the pointer is checked. I'd very much prefer only the length to be checked, so that converting to bool effectively checks for emptiness, not whether the base address is null (and non-empty slices starting at null are highly dubious IMO).

Reply:
Sorry, I was mistaken - DMD apparently uses (an IMO absolutely unintuitive) `slice.ptr || slice.length` for `cast(bool) slice`; LDC only checks the pointer. Seems totally untested, nice.

Reply:
> It's a persuasive idea, but I'd really prefer to just get away from 0 terminated strings.

Ok.

Reply:
> Sorry, I was mistaken - DMD apparently uses (an IMO absolutely unintuitive) `slice.ptr || slice.length` for `cast(bool) slice`

Yeah, reminds me of the horrible `String.IsNullOrEmpty(s)` in C#. LDC should better not behave differently, though.

Reply:
The test failure from a couple of hours ago was related to me not updating one of the format strings... It would certainly be nice to have slice-friendly error reporting